app/data/private/key.go

```diff
@@ -0,0 +1,53 @@
+package private
```

Comment:
I needed to work around a cyclic dependency with the mock packages which is why I introduced this sub-package. Open to suggestions of a better name or location. I've tried to make the package a natural part of the member names when references as is suggestd in go style guidelines